### PR TITLE
Smart exception

### PR DESCRIPTION
**Pull request by LukeBoyer**

### CODE REVIEW

File: api/src/main/java/com/codeforcommunity/exceptions/MalformedParameterException.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.codeforcommunity.rest.subrouter.FailureHandler;
+
+import io.vertx.ext.web.RoutingContext;
+
+public class MalformedParameterException extends RuntimeException implements HandledException {
+
+
+  @Override
+  public void callHandler(FailureHandler handler, RoutingContext ctx) {
```
Comment (liammoyn) [inline api/src/main/java/com/codeforcommunity/exceptions/MalformedParameterException.java]: Was this meant to be left blank?

File: api/src/main/java/com/codeforcommunity/rest/ApiRouter.java [hunk 2]
```diff
@@ -3,29 +3,24 @@
...
-import java.util.Optional;
 
 public class ApiRouter implements IRouter {
     private final CommonRouter commonRouter;
     private final NotesRouter notesRouter;
     private final AuthRouter authRouter;
 
-    public ApiRouter(INotesProcessor notesProcessor, IAuthProcessor authProcessor, JWTAuthorizer jwtAuthorizer) {
-        this.commonRouter = new CommonRouter(jwtAuthorizer);
+    public ApiRouter(INotesProcessor notesProcessor, IAuthProcessor authProcessor, JWTAuthorizer jwtAuthorizer, FailureHandler failureHandler) {
```
Comment (liammoyn) [inline api/src/main/java/com/codeforcommunity/rest/ApiRouter.java]: Considering FailureHandler doesn't need anything special I would probably just construct in the CommonHandler where it's being used

File: api/src/main/java/com/codeforcommunity/rest/subrouter/FailureHandler.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import com.codeforcommunity.exceptions.MissingHeaderException;
+import com.codeforcommunity.exceptions.MissingParameterException;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+import io.vertx.ext.web.RoutingContext;
+
+public class FailureHandler { //todo where should this file live?
```
Comment (liammoyn) [inline api/src/main/java/com/codeforcommunity/rest/subrouter/FailureHandler.java]: I think either here or in the /exceptions package is a fine place for it

File: api/src/main/java/com/codeforcommunity/exceptions/MalformedParameterException.java (hunk 1 above)
Comment (LukeBoyer) [reply api/src/main/java/com/codeforcommunity/exceptions/MalformedParameterException.java]: fixing this

File: api/src/main/java/com/codeforcommunity/rest/ApiRouter.java (hunk 2 above)
Comment (LukeBoyer) [reply api/src/main/java/com/codeforcommunity/rest/ApiRouter.java]: sounds good

**Pull request merged**